packages/remix-dev/config/routesConvention.ts

```diff
@@ -188,6 +240,8 @@ export function createRoutePath(partialRouteId: string): string | undefined {
 
   if (rawSegmentBuffer === "index" && result.endsWith("index")) {
     result = result.replace(/\/?index$/, "");
+  } else {
+    result = result.replace(/\/$/, "");
```

Comment:
Align route paths so we never include trailing slashes on `<Route path>` entries, which messes with ranking since /-delimited segments count in ranking

Reply:
Fixes #3014